uberfire-nio2-backport/uberfire-nio2-impls/uberfire-nio2-jgit/src/test/java/org/uberfire/java/nio/fs/jgit/util/extensions/JGitFSHooksTest.java

```diff
@@ -31,49 +34,52 @@
     @Test
     public void executeFSHooksTest() {
 
-        String fsName = "dora";
+        final String fsName = "dora";
+
         AtomicBoolean executedWithLambda = new AtomicBoolean(false);
 
-        FileSystemHooks.FileSystemHook<String> hook = spy(new FileSystemHooks.FileSystemHook<String>() {
+        FileSystemHooks.FileSystemHook hook = spy(new FileSystemHooks.FileSystemHook() {
             @Override
-            public void execute(String s) {
-                assertEquals(fsName, s);
+            public void execute(FileSystemHookExecutionContext context) {
+                assertEquals(fsName, context.getFsName());
             }
         });
 
-        FileSystemHooks.FileSystemHook<String> lambdaHook = s -> {
-            assertEquals(fsName, s);
+        FileSystemHooks.FileSystemHook lambdaHook = context -> {
+            assertEquals(fsName, context.getFsName());
             executedWithLambda.set(true);
         };
 
-        JGitFSHooks.executeFSHooks(hook, FileSystemHooks.ExternalUpdate, fsName);
-        JGitFSHooks.executeFSHooks(lambdaHook, FileSystemHooks.ExternalUpdate, fsName);
+        JGitFSHooks.executeFSHooks(hook, FileSystemHooks.ExternalUpdate, new FileSystemHookExecutionContext(fsName));
+        JGitFSHooks.executeFSHooks(lambdaHook, FileSystemHooks.ExternalUpdate, new FileSystemHookExecutionContext(fsName));
+
+        verify(hook).execute(any());
 
-        verify(hook).execute(fsName);
         assertTrue(executedWithLambda.get());
     }
 
     @Test
     public void executeFSHooksArrayTest() {
 
-        String fsName = "dora";
+        final String fsName = "dora";
+
         AtomicBoolean executedWithLambda = new AtomicBoolean(false);
 
-        FileSystemHooks.FileSystemHook<String> hook = spy(new FileSystemHooks.FileSystemHook<String>() {
+        FileSystemHooks.FileSystemHook hook = spy(new FileSystemHooks.FileSystemHook() {
             @Override
-            public void execute(String s) {
-                assertEquals(fsName, s);
+            public void execute(FileSystemHookExecutionContext context) {
+                assertEquals(fsName, context.getFsName());
             }
         });
 
-        FileSystemHooks.FileSystemHook<String> lambdaHook = s -> {
-            assertEquals(fsName, s);
+        FileSystemHooks.FileSystemHook lambdaHook = context -> {
+            assertEquals(fsName, context.getFsName());
             executedWithLambda.set(true);
         };
 
-        JGitFSHooks.executeFSHooks(Arrays.asList(hook, lambdaHook), FileSystemHooks.ExternalUpdate, fsName);
+        JGitFSHooks.executeFSHooks(Arrays.asList(hook, lambdaHook), FileSystemHooks.ExternalUpdate, new FileSystemHookExecutionContext(fsName));
 
-        verify(hook).execute(fsName);
+        verify(hook).execute(any());
```

Comment:
could we please also verify it was called with expected `FileSystemHookExecutionContext` ?

Reply:
done